src/block_ffm.rs

```diff
@@ -335,16 +412,21 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
             let ffmk: u32 = self.ffm_k;
             let ffmk_as_usize: usize = ffmk as usize;
 
+            let ffmk_end = ffmk_as_usize - ffmk_as_usize % STEP;
```

Comment:
I think doing &(step -1) vs % is like 6ns vs 1ns, random trivia

Reply:
Doing this only at the beginning of the loop :) so we will not save much :)